lib/SPIRV/OCLToSPIRV.cpp

```diff
@@ -1419,6 +1436,105 @@ void OCLToSPIRVBase::visitCallDot(CallInst *CI) {
+  }
+  if (CI->getNumArgOperands() > 3) {
+    Args.push_back(CI->getOperand(3));
+  }
+  if (CI->getOperand(0)->getType()->isVectorTy()) {
+    if (IsDot) {
+      // dot(char4, char4) @_Z3dotDv4_cS_
+      // dot(char4, uchar4) @_Z3dotDv4_cDv4_h
+      // dot(uchar4, char4) @_Z3dotDv4_hDv4_c
+      // dot(uchar4, uchar4) @_Z3dotDv4_hS_
+      // or
+      // dot(short2, short2) @_Z3dotDv2_sS_
+      // dot(short2, ushort2) @_Z3dotDv2_sDv2_t
+      // dot(ushort2, short2) @_Z3dotDv2_tDv2_s
+      // dot(ushort2, ushort2) @_Z3dotDv2_tS_
+      if (MangledName[MangledName.size() - 1] == '_') {
+        IsFirstSigned = ((MangledName[MangledName.size() - 3] == 'c') || 
+                        (MangledName[MangledName.size() - 3] == 's'));
+        IsSecondSigned = IsFirstSigned;
+      } else {
+        IsFirstSigned = ((MangledName[MangledName.size() - 6] == 'c') ||
+                        (MangledName[MangledName.size() - 6] == 's'));
+        IsSecondSigned = ((MangledName[MangledName.size() - 1] == 'c') ||
+                         (MangledName[MangledName.size() - 1] == 's'));
+      }
+    } else {
+      // dot_acc_sat(char4, char4, int) @_Z11dot_acc_satDv4_cS_i
+      // dot_acc_sat(char4, uchar4, int) @_Z11dot_acc_satDv4_cDv4_hi
+      // dot_acc_sat(uchar4, char4, int) @_Z11dot_acc_satDv4_hDv4_ci
+      // dot_acc_sat(uchar4, uchar4, uint) @_Z11dot_acc_satDv4_hS_j
+      // or
+      // dot_acc_sat(short2, short2, int) @_Z11dot_acc_satDv4_sS_i
+      // dot_acc_sat(short2, ushort2, int) @_Z11dot_acc_satDv4_sDv4_ti
+      // dot_acc_sat(ushort2, short2, int) @_Z11dot_acc_satDv4_tDv4_si
+      // dot_acc_sat(ushort2, ushort2, uint) @_Z11dot_acc_satDv4_tS_j
+      IsFirstSigned = ((MangledName[19] == 'c') || (MangledName[19] == 's'));
+      IsSecondSigned = (MangledName[20] == 'S'
+                            ? IsFirstSigned
+                            : ((MangledName[MangledName.size() - 2] == 'c') ||
+                               (MangledName[MangledName.size() - 2] == 's')));
+    }
+  } else {
+      // for packed format
+      // dot(int, int, int) @_Z3dotiii
+      // dot(int, uint, int) @_Z3dotiji
+      // dot(uint, int, int) @_Z3dotjii
+      // dot(uint, uint, int) @_Z3dotjji
+      // or
+      // dot_acc_sat(int, int, int, int) @_Z11dot_acc_satiiii
+      // dot_acc_sat(int, uint, int, int) @_Z11dot_acc_satijii
+      // dot_acc_sat(uint, int, int, int) @_Z11dot_acc_satjiii
+      // dot_acc_sat(uint, uint, int, int) @_Z11dot_acc_satjjii 
+      IsFirstSigned = (IsDot) ? (MangledName[MangledName.size() - 3] == 'i')
+                              : (MangledName[MangledName.size() - 4] == 'i');
+      IsSecondSigned = (IsDot) ? (MangledName[MangledName.size() - 2] == 'i')
+                               : (MangledName[MangledName.size() - 3] == 'i');
+  }
+  std::string NamePref =
+      (IsFirstSigned != IsSecondSigned ? "SU" : ((IsFirstSigned) ? "S" : "U"));
+  std::string FinName = "_Z" +
```

Comment:
It might be better to use `mutateCallInstSPIRV` inside this method, which will handle the mangling for you.